static/js/hooks/search.ts

```diff
@@ -76,3 +79,50 @@ export function useURLParamFilter<LParams extends ListingParamsMinimum>(
     listingParams
   }
 }
+
+/**
+ * The location interface used by react-router
+ */
+type RLocation = ReturnType<typeof useLocation>
+
+/**
+ * Increment the `offset` value for a route location.
+ * @param location The location to clone + modify
+ * @param count The total count of page items
+ * @param increment Amount by which to increment the current offset
+ * @returns A copy of `location` with new offset based on count + increment.
+ */
+const offsetLocation = (
+  location: RLocation,
+  count: number,
+  increment: number
+): RLocation | null => {
+  const params = new URLSearchParams(location.search)
+  const offset = Number(params.get("offset"))
+  if (offset === 0 && increment < 0) return null
+  const newOffset = clamp(offset + increment, 0, count)
+  if (newOffset >= count) return null
+  params.set("offset", String(newOffset))
+  return {
+    ...location,
+    search: params.toString()
+  }
+}
+
+/**
+ * A hook that returns the frontend routes for "Previous" and "Next" buttons.
+ * The `next` and `previous` routes are either null or they are the same as the
+ * current route with a different `offset` query parameter.
+ */
+export const usePagination = (
+  count: number,
+  pageSize: number = WEBSITE_CONTENT_PAGE_SIZE
+): {
+  previous: RLocation | null
+  next: RLocation | null
+} => {
+  const location = useLocation()
+  const previous = offsetLocation(location, count, -pageSize)
+  const next = offsetLocation(location, count, +pageSize)
+  return { previous, next }
```

Comment:
**Comment:** The API responses from our listings APIs include ` { count, previous, next }`. Of that info, we're only using `count` and not previous/next for the frontend pagination.

I did consider basing this logic off of the API responses. One challenge with that is that the API routes do not map directly to our frontend routes. I decided logic of the form "keep frontend url exactly the same except for offset parameter" was preferable to trying to convert API routes to frontend routes since that varies about by scenario (content listing, site listing).